xs2a-adapter-rest-server/src/main/java/de/adorsys/xs2a/adapter/controller/AspspModifyController.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018-2018 adorsys GmbH & Co KG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.adorsys.xs2a.adapter.controller;
+
+import de.adorsys.xs2a.adapter.mapper.AspspMapper;
+import de.adorsys.xs2a.adapter.model.AspspTO;
+import de.adorsys.xs2a.adapter.service.AspspModifyRepository;
+import de.adorsys.xs2a.adapter.service.model.Aspsp;
+import org.mapstruct.factory.Mappers;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+
+@Profile("dev")
+@RestController
+public class AspspModifyController {
+    static final String ASPSP_ID = "{aspspId}";
+    static final String V1_ASPSP_BY_ID = AspspService.V1_APSPS + "/" + ASPSP_ID;
+    private final AspspModifyRepository aspspModifyRepository;
+    private final AspspMapper aspspMapper = Mappers.getMapper(AspspMapper.class);
+
+    public AspspModifyController(AspspModifyRepository aspspModifyRepository) {
+        this.aspspModifyRepository = aspspModifyRepository;
+    }
+
+    @PostMapping(AspspService.V1_APSPS)
+    ResponseEntity<AspspTO> addAspsp(@RequestBody AspspTO to) {
+        Aspsp aspsp = aspspModifyRepository.create(aspspMapper.toAspsp(to));
+        String uri = V1_ASPSP_BY_ID.replace(ASPSP_ID, aspsp.getId());
+        return ResponseEntity.created(URI.create(uri)).body(aspspMapper.toAspspTO(aspsp));
+    }
+
+    @PutMapping(AspspService.V1_APSPS)
+    ResponseEntity<AspspTO> updateAspsp(@RequestBody AspspTO aspsp){
```

Comment:
whitespace problems in this file as well

Reply:
thanks, fixed